pr

L-as opened the pull request:
- [x] anabatic
 - [ ] bootstrap
 - [x] bora
 - [x] coloquinte
 - [x] crlcore
 - [x] cumulus
 - [ ] documentation
 - [x] equinox
 - [x] etesian
 - [x] flute
 - [x] hurricane
 - [x] ispd
 - [x] karakaze
 - [x] katabatic
 - [x] katana
 - [x] kite
 - [x] knik
 - [x] lefdef
 - [x] mauka
 - [x] metis
 - [x] nimbus
 - [x] oroshi
 - [x] solstice
 - [x] stratus1
 - [x] tutorial
 - [x] unicorn
 - [x] unittests
 - [x] vlsisapd

L-as committed:
wip

L-as committed:
bad hack

MatthewCroughan commented on lefdef/src/lef/lef/CMakeLists.txt:
```diff
@@ -61,8 +61,8 @@
                                COMMAND  ${BISON_EXECUTABLE} -v -p lefyy -d ${LefParserGrammar} -o lef.tab.cpp 
-                               COMMAND  mv lef.tab.hpp ${LEFDEF_SOURCE_DIR}/src/lef/lef/lef.tab.h
-                              #COMMAND  mv lef.tab.cpp ${LEFDEF_SOURCE_DIR}/src/lef/lef/lef.tab.cpp
+                               COMMAND  cp lef.tab.hpp ${LEFDEF_SOURCE_DIR}/src/lef/lef/lef.tab.h
+                               #COMMAND  mv lef.tab.cpp ${LEFDEF_SOURCE_DIR}/src/lef/lef/lef.tab.cpp
```
How come you are making changes to the `lefdef/src/lef/lef/CMakeLists.txt` file? Was this just for testing? Or do you intend to upstream this change in addition to the flake? In my mind, the correct way to do this is a patch.

L-as replied:
I'm not sure what I'm going to do with this flake: https://github.com/ngi-nix/summer-of-nix/discussions/44

In any case, I can convert the commit to a diff later (though the change would be useful upstream anyway).

L-as committed:
Got it working?

L-as committed:
Use CMake 2.8

L-as committed:
temporary commit

L-as committed:
Get vlsisapd and bootstrap building!

L-as committed:
hurricane builds

L-as commented:
I've made an overview of the components that need to be packaged.

L-as committed:
coloquinte and lefdef

L-as committed:
Cleanup

L-as committed:
crlcore work

L-as committed:
crlcore

L-as committed:
cumulus

L-as committed:
flute

DieracDelta commented on nix/coloquinte.nix:
```diff
@@ -0,0 +1,18 @@
+
+{ lib, stdenv, python2, cmake, boost, qt4, ninja, coriolis-bootstrap }:
+
+let boostWithPython = boost.override { enablePython = true; python = python2; }; in
```
They depend on python 2? That's mildly concerning. Saw the readme...

L-as replied:
Libre-SoC seems to maybe use Python 3..? https://libre-soc.org/HDL_workflow/coriolis2/

I guess we'll find out when we try actually using it.

L-as committed:
etesian and anabatic

L-as committed:
katana

L-as committed:
bora

L-as committed:
equinox

L-as committed:
knik

L-as committed:
katabatic

L-as committed:
ispd work

L-as committed:
include/coriolis -> include/coriolis2

L-as committed:
ispd

L-as committed:
karakaze

L-as committed:
kite

L-as committed:
nimbus

L-as committed:
metis, mauka

L-as committed:
oroshi

L-as committed:
solstice

L-as committed:
stratus1

L-as committed:
tutorial

L-as committed:
Build documentation

L-as committed:
unicorn

L-as committed:
documentation

L-as committed:
unittests

L-as commented:
It's all packaged now, all that's left is testing it now.

L-as committed:
-DBUILD_DOC=ON

L-as committed:
Set version and default package

L-as commented:
It doesn't run, but I'm not going to debug this now.

DieracDelta commented on nix/mauka.nix:
```diff
@@ -0,0 +1,24 @@
+
+  inherit version;
+
+  meta = meta // { broken = true; };
```
is this broken?

L-as replied:
Yep, it seems to be deprecated. It needs symbols from some other library in Coriolis (Hurricane IIRC) that aren't defined in the first place, so I assume it's neglected.

DieracDelta replied:
ahh gotcha, thanks

L-as committed:
Fix linking with boost python

L-as committed:
Make vlsisapd accessible through Python!

L-as committed:
Add generic helper

L-as committed:
hurricane

L-as committed:
Try to get crlcore working

L-as committed:
flute

L-as committed:
etesian, anabatic

L-as committed:
katana

L-as committed:
unicorn, tutorial, kite, katabatic, bora

L-as commented:
I think it might work partially now. The `cgt` binary likely still doesn't run (haven't tested it), but I can import the Python extension modules from `unicorn`, etc.

L-as committed:
Add devShells

L-as committed:
crlcore

L-as committed:
Fix licenses

L-as committed:
Add license for Nix code

L-as committed:
Try to get Qt support to work

L-as commented:
https://github.com/NixOS/nixpkgs/pull/123263

L-as committed:
Use pyqt4 4.12.1

L-as committed:
Fix missing import in crlcore

L-as committed:
Set CORIOLIS_TOP for crlcore check

L-as committed:
Get oroshi working

L-as committed:
Make cgt work a bit more

L-as committed:
Use OpenMP

L-as commented:
Trying to get tests at https://gitlab.lip6.fr/vlsi-eda/alliance-check-toolkit working.

L-as committed:
Run unit tests

L-as committed:
Add initial code for running alliance checks

L-as committed:
Include yosys

L-as commented:
Fixed Alliance support in Nixpkgs: https://github.com/L-as/nixpkgs/commits/alliance

Perhaps @DieracDelta wants to take a look at it before I present it upstream? (I will also have to test it with alliance-check-toolkit before that though.)

L-as commented:
Sent Jean-Paul Chaput an e-mail about perhaps upstreaming this work.

L-as committed:
Use Nixpkgs with Alliance

L-as committed:
Get alliance check toolkit to fail

L-as committed:
Fix devShell

L-as committed:
Get further in alliance tests

L-as committed:
Fix cumulus

L-as committed:
Some checks pass!

L-as committed:
Add combined

L-as committed:
fix karakaze

L-as committed:
Keep backward compatibility for cumulus users

L-as committed:
Fix stratus1

L-as committed:
Get adder/cmos test working!

L-as committed:
Don't try to catch import error in cgt

L-as committed:
Run all tests (some fail)

L-as committed:
Make all checks work

DieracDelta commented on flake.nix:
```diff
@@ -0,0 +1,113 @@
+        }
+      );
+
+      defaultPackage = forAllSystems (system: self.packages.${system}.unicorn);
```
```suggestion
      # CORIOLIS_TOP must be set before execution
      # example: CORIOLIS_TOP="$(mktemp -d)" && ./result/bin/cgt
      defaultPackage = forAllSystems (system: self.packages.${system}.unicorn);
```

DieracDelta commented on nix/alliance-check-toolkit.nix:
```diff
@@ -0,0 +1,28 @@
+{ alliance-src, ... }:
+
+{ stdenv, coriolis-combined, coriolis-unicorn, coriolis-cumulus
+, python2Packages, alliance, yosys }:
+
+stdenv.mkDerivation {
+  pname = "alliance-check-toolkit";
+  version = builtins.substring 0 8 alliance-src.lastModifiedDate;
+  src = alliance-src;
+
+  nativeBuildInputs = [ alliance coriolis-unicorn coriolis-cumulus yosys ];
+
+  YOSYS_TOP = "${yosys}";
+  ALLIANCE_TOP = "${alliance}";
+  CORIOLIS_TOP = "${coriolis-combined}";
+  # The user configuration for verhaegs is empty,
+  # which is why we use it.
+  USER = "verhaegs";
+
+  postPatch = "patchShebangs .";
+  configurePhase = "export CHECK_TOOLKIT=\"$PWD\"";
+  buildPhase = ''
+    cd benchs
+    ../bin/go.sh
+  '';
+  installPhase = "touch $out";
+  fixupPhase = "true";
+}
```
Is this meant as a check? As in "if it builds, the check passes"

L-as replied:
Yep. I'm going to make it a check in the flake too.

DieracDelta replied:
Yeah, that makes sense. The resulting `result` file isn't marked as executable for me (and by the looks of it is an empty file).

L-as committed:
Add comment about CORIOLIS_TOP

DieracDelta commented on nix/tutorial.nix:
```diff
@@ -0,0 +1,18 @@
+    nativeBuildInputs = [ doxygen ];
+    pythonImportsCheck = [ "Tutorial" ];
+  };
+in generic pkg
```
I'm getting a failure here: 
```
❯ ./tutorial
[ERROR] The <Cfg> python module or symbol cannot be loaded.
        Please check the integrity of the <coriolis> package.
[ERROR] The <No module named Cfg> shared library cannot be loaded.
        Under RHEL 6, you must be under devtoolset-2.
        (scl enable devtoolset-2 bash)
```
I'm guessing this is only used as a dependency of unicorn?

L-as replied:
Yeah, I'm not sure how you're supposed to use it actually.

DieracDelta commented on nix/unittests.nix:
```diff
@@ -0,0 +1,32 @@
+{ version, meta }:
```
might be worth adding this package as a check as well

L-as replied:
Will do.

L-as committed:
Make alliance-check-toolkit a check

L-as committed:
Make unittests a check

L-as committed:
Add nix/libresoc-experiments9.nix

L-as committed:
Fix comment

L-as committed:
Fix unittests

L-as committed:
LibreSoC experiment work

L-as committed:
Try to get libresoc working

L-as committed:
Make Libre-SoC almost work

L-as committed:
Use older version of yosys for Libre-SoC

DieracDelta commented:
Well done!! LGTM with one caveat. Very clean stuff IMO. We should definitely try to upstream if possible. I think this could make a huge difference in the usability of coriolis.

I have one question: when running the tool, I can't seem to get the tutorials to run. Any ideas on how we could get this to work/why this doesn't run?

L-as commented:
I'm not sure, perhaps 1.4 in https://libre-soc.org/HDL_workflow/coriolis2/ is how to use it? Perhaps it has to be used through `cgt`?

DieracDelta commented:
To be a bit clearer on the steps involved:
```nix
nix build .
export CORIOLIS_TOP="$(mktemp -d)"
./result/bin/cgt
```

Then I select either of the two tutorials presented by the dropdown:
![image](https://user-images.githubusercontent.com/13730968/132247370-273f5969-9545-4708-b51e-ee25cc95f9fb.png)
![image](https://user-images.githubusercontent.com/13730968/132247391-d1de4b77-d352-46f7-af79-3474ea7d24af.png)

This drops the following warning:
```
[WARNING] Unicorn system configuration file:
  </tmp/tmp.Czu4VZvnCe/lib/python2.7/site-packages/unicorn/unicornInit.py> not found.
  ```
I think this is the step process they're describing in that readme you've linked to.

DieracDelta commented:
It looks like `unicornInit.py` is in `result`. When I export `CORIOLIS_TOP` to be `result`, then cgt fails to start. I get the following errors:
```
❯ ./result/bin/cgt
  o  Locating configuration directory:
     - "/home/jrestivo/SoN/reviews/coriolis/result/etc/coriolis2"
     - ".../lib/python2.7/site-packages/symbolic/cmos/technology.py".
     - ".../lib/python2.7/site-packages/symbolic/cmos/display.py".
     - ".../lib/python2.7/site-packages/symbolic/cmos/analog.py".
     - ".../lib/python2.7/site-packages/symbolic/cmos/alliance.py".
     - ".../lib/python2.7/site-packages/symbolic/cmos/etesian.py".
     - ".../lib/python2.7/site-packages/symbolic/cmos/kite.py".
     - ".../lib/python2.7/site-packages/symbolic/cmos/plugins.py".
     - ".../lib/python2.7/site-packages/symbolic/cmos/stratus1.py".
```

 which makes sense. It looks like they're using relative paths. This might be entirely red-herring of what the actual reason is, though.

L-as commented:
@DieracDelta From what I can tell, you're supposed to have some settings in the current directory. What happens if you copy it to your cwd?

L-as committed:
libresoc: Use recon experiment instead

L-as commented:
Note to myself: I need to add the license attributes for the packages.

L-as committed:
Remove unneeded input

DieracDelta commented:
@L-as I didn't have much luck trying that (same thing).

I don't think getting the tutorial to work is super high priority just as long as it can do its job for libre-SoC. The code quality is high and it seems to run fine. 

Did you have any luck contacting upstream?

L-as commented:
I haven't contacted them since Jean-Paul Chaput told me he was taking a vacation for 2 weeks. I should probably contact him again.

aaronjanse commented:
Would it be possible to do the `coriolis -> coriolis2` rename programmatically (e.g. via something like `sed`) instead of writing a large patch?

L-as commented:
Why would you rename it?

aaronjanse commented:
Oh, I was referring to the diff. For example:

https://github.com/ngi-nix/coriolis/pull/1/files#diff-0daba194b4ca6ffabf28a934928d4fbff89dda356c20809e86581c563e0d8fe9R21

L-as commented:
Ah, I mean yeah, I could do a sed instead, but the problem is that this is actually an upstream bug, i.e. the CMake files are broken. When I eventually upstream this, I imagine they want the fixed CMake files too.

aaronjanse commented:
Oh got it. Sounds good

L-as committed:
Fix licenses

L-as commented:
I merged with upstream, not sure if it still works lol

aaronjanse commented:
The `flake.lock` seems to [reference a commit that does not exist](https://github.com/ngi-nix/coriolis/blob/6c44a7a8a1293fa32d51bfad4b37d766eb771679/flake.lock#L36-L43), causing the following error:
```
error: unable to download 'https://api.github.com/repos/L-as/nixpkgs/tarball/11f9ff27bc66bf653ccbbfae1412e6f5705e4a5c': HTTP error 404 ('')

       response body:

       404: Not Found
(use '--show-trace' to show detailed location information)
```

L-as commented:
Ah! I've deleted the branch and forgot to update the input here. My PRs have been merged now after all.

L-as commented:
I'm in contact with upstream and trying to merge it upstream so I'll fix this.

L-as committed:
Adapt Nix code to Python 3 switch.

This commit also reduces the diff by not removing
some (unnecessary) error handling.

L-as committed:
nix: Update dependencies

L-as committed:
nix: Fix missing imports

L-as committed:
nix: Use Python 3

L-as committed:
nix: Remove dependencies on vlsisapd

L-as commented:
I made a PR: https://gitlab.lip6.fr/vlsi-eda/coriolis/-/merge_requests/8

L-as committed:
nix: Disable tests temporarily

L-as committed:
Ignore /result (from nix-build)

L-as committed:
nix: add compatibility files

L-as committed:
nix: Remove unnecessary `ln` from libresoc test

L-as commented:
This has been merged! https://gitlab.lip6.fr/vlsi-eda/coriolis/-/merge_requests/8
